### ISSUE TITLE
Postprocessing MAX_PARQUET_MEMORY too high?

### ISSUE THREAD

**Issue by spolloni** [NREL/buildstockbatch#210]
Hello there,

I'm encountering nasty Dask errors with the current `MAX_PARQUET_MEMORY` setting when postprocessing results. When the codes reaches the parquet timeseries step, several warnings are printed:
```
distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.19 GB -- Worker memory limit: 6.33 GB  
```
Then eventually the script fails with the following error:
```
distributed.scheduler.KilledWorker: ("('from-delayed-read_and_concat_enduse_timeseries_parquet-set_index-9bdd5f8bca01636a0a0ff8d724e3093a-index-5ad59557d04-1781', 25)", <Worker 'tcp://127.0.0.1:34983', name: 4, memory: 0, processing: 4>)
```

Looking at the code below, I wonder if `MAX_PARQUET_MEMORY` should be determined as a function of how many CPUs dask will utilize and the total available RAM? 

https://github.com/NREL/buildstockbatch/blob/94597284e6a61521bd94358f015501b701f3eb1c/buildstockbatch/postprocessing.py#L335-L338

In my case, reducing this parameter from `1E9` to `2E8` avoided any further failures.

**Comment by nmerket** [comment]
That in-memory size of the partition has been more or less tuned to generate parquet files that are "optimal" size for querying in AWS Athena. It can require a significant amount of memory to deal with them, though. If I'm being honest we've never really performance tuned this application to run at any meaningful scale on a single machine. When we run the dask cluster on NREL's supercomputer, Eagle, we run a dask cluster with several of the [192GB nodes](https://www.nrel.gov/hpc/eagle-system-configuration.html) to get enough memory. And honestly, we've still had trouble when we're outputing a lot of variables. @rajeee has been doing work recently to get that working better. And on AWS, we run the dask cluster on EMR by default with workers in the [`r5.` class](https://aws.amazon.com/ec2/instance-types/r5/), which have a higher memory to CPU ratio.

That being said, I could see opening up that variable to be an optional user-specified value.

**Comment by spolloni** [comment]
> That in-memory size of the partition has been more or less tuned to generate parquet files that are "optimal" size for querying in AWS Athena.

That makes sense. 

Have you considered using the Dask DataFrame [repartition](https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.DataFrame.repartition) method to handle partitioning?  Seems like the `read_and_concat_enduse_timeseries_parquet` function requires holding all of the concatenated partition in memory. I have no idea how Dask works internally, but maybe it's able to manage memory a bit better?  [Here](https://github.com/spolloni/buildstockbatch/commit/fd76dbc38d1949476463d6ec49d2918c65d13309) is roughly how it would look like. 

I haven't tested this extensively and have no idea how it'll behave on a cluster, but on my machine I was able to achieve larger partition sizes without Dask complaining too much. Also note that the `repartition` method has a `partition_size` argument, which could avoid having to calculate `npartitions` altogether if the goal is simply to hit the ~1GB target.

**Comment by nmerket** [comment]
@spolloni Sorry this took so long to get back to you. We looked at repartition at one point. The dask scheduler starts to choke on many (100k - 1m) partitions. Maybe they've improved in more recent versions. That function actually reduces a lot of the overhead, which is why we do the concats there. In @rajeee's latest fix we actually did away with dask dataframes altogether because of the instability.
